tests/e2e/utils/reset.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * Internal dependencies
+ */
+import { switchUserToAdmin, switchUserToTest, visitAdminPage } from '@wordpress/e2e-test-utils';
+
+const PLUGIN_SLUG = 'e2e-tests-reset-plugin';
+
+/**
+ * Reset Site Kit using utility plugin.
+ */
+export async function resetSiteKit() {
```

Comment:
There's a `activatePlugin` utility that can be used for this.

So `resetSiteKit()` basically turns into `await activatePlugin( 'e2e-tests-reset-plugin' )`

Reply:
Yes, but it doesn't work with the reset plugin because it deactivates itself after performing the reset which causes the selector it's waiting for (the deactivate plugin link) to fail. This utility is based on core's `activatePlugin`.

Reply:
So why does the plugin deactivate itself then? What's the benefit of that?

Activating and deactivating the reset plugin in tests seems more explicit and easier to grasp.

Reply:
There's no reason for the plugin to remain active, and leaving it active could result in the Reset running more than once. It's also saves an additional page load to not require deactivating it via the UI.

Apart from that, I think it's more expressive to have a utility for resetting rather than requiring the user to know that activating the reset plugin resets Site Kit, which is not explicit. Activating and deactivating the utility is more explicit about the plugin activation (that's more of an implementation detail IMO), not about resetting Site Kit.

Reply:
> There's no reason for the plugin to remain active, and leaving it active could result in the Reset running more than once

The reset could happen in a plugin activation hook. Seems to work quite well, and I implemented that in ccd3d600fc03eabb21e76ed2a57c74bc10c541ea to get a better feeling for it.

I get your point about the implementation detail though. The plugin activation could still be put in a `resetSiteKit` utility if that makes it more expressive.

Then, the plugin deactivation could even be handled in `tests/e2e/config/bootstrap.js`'s `afterEach` method.

Reply:
> The reset could happen in a plugin activation hook.

Yes, it could work using an activation hook as well. One of the reasons I didn't choose this initially, and hooked it on `googlesitekit_init` instead was to ensure Site Kit was active and loaded.

> I get your point about the implementation detail though. The plugin activation could still be put in a resetSiteKit utility if that makes it more expressive.

The deactivation could be put in the same utility as well; there's no reason to leave that as a manual step to be handled in tear down. In which case, why not make it self-deactivating? 😄 

>  I implemented that in ccd3d60 to get a better feeling for it.

I have the code for the utility plugins isolated in their own branch right now so that these can be pulled into whichever tests need them without causing merge conflicts later since none of them have been merged yet (there's at least one other test that uses it right now). Can we move any changes to the reset utility to the `feature/e2e-utility-reset` branch and then merge it back in here?

Reply:
> to ensure Site Kit was active and loaded.

Valid thought, but I'd say it's fair to assume that the Site Kit plugin is always active and running during E2E tests (otherwise the tests will fail anyway). The only exception is the already existing test for the plugin activation.

> Can we move any changes to the reset utility to the `feature/e2e-utility-reset` branch and then merge it back in here?

Sure.

> In which case, why not make it self-deactivating

```js
/**
 * Reset Site Kit using utility plugin.
 */
export async function resetSiteKit() {
	await switchUserToAdmin();
	await activatePlugin( 'e2e-tests-reset-plugin' );
	await deactivatePlugin( 'e2e-tests-reset-plugin' );
	await switchUserToTest();
}
```

Has the exact same effect then 😄 No strong opinion on this though, otherwise we start bikeshedding :)

Reply:
> Has the exact same effect then 😄

Sounds good to me. Like I said, I made it self-deactivating partly to save an extra page load for a utility that will be used quite a bit in the tests. I don't mind changing it to use core utilities internally rather than being a custom one. I think it can even be simplified a bit more 😄  

```js
export async function resetSiteKit() {
	await activatePlugin( 'e2e-tests-reset-plugin' );
	await deactivatePlugin( 'e2e-tests-reset-plugin' );
}
```